### DISCUSSION TITLE
Show Toast within Synchronous Method

### DISCUSSION THREAD

**Discussion by Orgbrat** [CommunityToolkit/Maui#487] [Q&A] [answered]
What is the correct way to show a Toast that is inside of a synchronous method?

**Comment by VladislavAntonyuk** [comment, marked as answer]
```csharp
async void ShowToast()
{
    await Toast.Make(title).Show();
}
```

**Reply by Orgbrat** [reply]
Hi @VladislavAntonyuk  and @brminnick ,
      I need to show the Toast from inside a synchronous method that is called from inside the class constructor.  So no async/ await allowed...  Was trying to stay away from a Task.Run() kinda thing or should I just do the Task.Run()?

**Reply by TheCodeTraveler** [reply]
```csharp
class MyContentPage : ContentPage
{
  public MyContentPage()
  {
      ShowToast();
  }

  async void ShowToast()
  {
      await Toast.Make(title).Show();
  }
}
```

**Reply by TheCodeTraveler** [reply]
Alternatively, you could use the [`.SafeFireAndForget()` extension method](https://github.com/brminnick/AsyncAwaitBestPractices#safefireandforget) in my `AsyncAwaitBestPractices` NuGet Package:
https://www.nuget.org/packages/AsyncAwaitBestPractices/

```csharp
class MyContentPage : ContentPage
{
  public MyContentPage()
  {
      Toast.Make(”Hello”).Show().SafeFireAndForget();
  }
}
```

**Reply by Orgbrat** [reply]
Ok @brminnick you convinced me. I added AsyncAwaitBestPractices nuget to my Maui application.

**Comment by CliffAgius** [comment]
Closed as answered
